client/src/app/api/model/instance.ts

```diff
@@ -28,21 +28,31 @@ export const ComponentTypeEnum = {
 
 export interface Instance {
   id?: number;
-  host: string;
-  portNumber: number;
-  name: string;
+  host?: string;
+  portNumber?: number;
+  name?: string;
   /**
    * Component Type
    */
-  componentType: ComponentType;
+  componentType?: Instance.ComponentTypeEnum;
   dockerId?: string;
   /**
    * State of the instance
    */
-  state?: StateEnum;
+  instanceState?: StateEnum;
   labels?: Array<string>;
 }
 
+export namespace Instance {
```

Comment:
why did you reintroduce the instance namespace? it was removed intentionally before and is currently no part of the develop branch.

Reply:
It was removed without our knowledge, even though we used it in much of our code.